fastlane_core/lib/fastlane_core/itunes_transporter.rb

```diff
@@ -227,4 +156,155 @@ def shell_escaped_password(password)
+    # the #{CredentialsManager::AccountManager}
+    def initialize(user = nil, password = nil, avoid_shell_script = false)
+      data = CredentialsManager::AccountManager.new(user: user, password: password)
+      @user = data.user
+      @password = data.password
+      @transporter_executor = avoid_shell_script ? JavaTransporterExecutor.new : ShellScriptTransporterExecutor.new
+    end
+
+    # Downloads the latest version of the app metadata package from iTC.
+    # @param app_id [Integer] The unique App ID
+    # @param dir [String] the path in which the package file should be stored
+    # @return (Bool) True if everything worked fine
+    # @raise [Deliver::TransporterTransferError] when something went wrong
+    #   when transfering
+    def download(app_id, dir = nil)
+      dir ||= "/tmp"
+
+      UI.message("Going to download app metadata from iTunes Connect")
+      command = @transporter_executor.build_download_command(@user, @password, app_id, dir)
+      UI.verbose(@transporter_executor.build_download_command(@user, 'YourPassword', app_id, dir))
+
+      result = @transporter_executor.execute(command, ItunesTransporter.hide_transporter_output?)
+      return result if Helper.is_test?
+
+      itmsp_path = File.join(dir, "#{app_id}.itmsp")
+      successful = result && File.directory?(itmsp_path)
+
+      if successful
+        UI.success("✅ Successfully downloaded the latest package from iTunes Connect to #{itmsp_path}")
+      else
+        handle_error(@password)
+      end
+
+      successful
+    end
+
+    # Uploads the modified package back to iTunes Connect
+    # @param app_id [Integer] The unique App ID
+    # @param dir [String] the path in which the package file is located
+    # @return (Bool) True if everything worked fine
+    # @raise [Deliver::TransporterTransferError] when something went wrong
+    #   when transfering
+    def upload(app_id, dir)
+      dir = File.join(dir, "#{app_id}.itmsp")
+
+      UI.message("Going to upload updated app to iTunes Connect")
+      UI.success("This might take a few minutes, please don't interrupt the script")
+
+      command = @transporter_executor.build_upload_command(@user, @password, dir)
+
+      UI.verbose(@transporter_executor.build_upload_command(@user, 'YourPassword', dir))
+
+      result = @transporter_executor.execute(command, ItunesTransporter.hide_transporter_output?)
+
+      if result
+        UI.success("-" * 102)
+        UI.success("Successfully uploaded package to iTunes Connect. It might take a few minutes until it's visible online.")
+        UI.success("-" * 102)
+
+        FileUtils.rm_rf(dir) unless Helper.is_test? # we don't need the package any more, since the upload was successful
+      else
+        handle_error(@password)
+      end
+
+      result
+    end
+
+    private
+
+    def handle_error(password)
+      # rubocop:disable Style/CaseEquality
+      unless /^[0-9a-zA-Z\.\$\_]*$/ === password
+        UI.error([
```

Comment:
Shouldn't this be fixed with this PR? Or do we still want to show it, just to be sure?